### DISCUSSION TITLE
Auth not being properly loaded from a cookie

### DISCUSSION THREAD

**Discussion by joeinnes** [pocketbase/pocketbase#3919] [Q&A] [answered]
Hi folks, scratching my head a bit on this one.

I've got a SvelteKit app which uses PocketBase. I've implemented more or less the `hooks.server.js` that the docs suggest. But for some reason, it seems that the `authStore` is not loading from the cookie.

Here's the bit that's got me scratching my head.

```js
const cookie = event.request.headers.get('cookie');
console.log(cookie);

// load the store data from the request cookie string
event.locals.pb.authStore.loadFromCookie(cookie || '');

console.log(event.locals.pb.authStore);
```

In my console, I see the following:

```js
pb-auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJjb2xsZWN0aW9uSWQiOiJfcGJfdXNlcnNfYXV0aF8iLCJleHAiOjE3MDM0OTQyOTEsImlkIjoiM2dnMWNpNGRoNGgzdTlwIiwidHlwZSI6ImF1dGhSZWNvcmQifQ.fCcGItVh52DC0nDxQ0tKxidIcvFaDVai17PGU0zfU7M
LocalAuthStore {
  baseToken: '',
  baseModel: null,
  _onChangeCallbacks: [],
  storageFallback: { pocketbase_auth: { token: '', model: null } },
  storageKey: 'pocketbase_auth'
}
```

As far as I can tell, the token is valid, but for some reason, it's not being loaded into the auth store.

I've started and restarted the app, deleted the cookie from the browser and tried again, invalidated all the tokens from PocketBase, but no joy.

Can anyone see what's going wrong?

**Comment by ganigeorgiev** [comment, marked as answer]
At first glance the cookie key doesn't seem to match with the default one - `pb-auth` vs `pb_auth` (default).

If you want to use a custom cookie key, you can pass it as a second argument to the `loadFromCookie(cookie, key)` method.

If that's not the case, please provide a more detailed code sample or ideally a **minimal** reproducible repo.

**Comment by joeinnes** [comment]
Thanks! That'll do it.
